react/Dialog/DialogEffects.ts

```diff
@@ -120,23 +122,73 @@ const makeCaller = (
 const getRootModal = (): HTMLElement | null => {
   const modals = document.querySelectorAll(DOMStrings.DialogClass)
 
-  return modals.length > 0 ? (modals[0] as HTMLElement) : null
+  return modals.length > 1 ? (modals[0] as HTMLElement) : null
 }
 
-const useHook = (fullscreen?: boolean): void => {
+const useHook = (open: boolean, fullscreen?: boolean): void => {
   const theme = useTheme()
   const cozybar = document.querySelector(DOMStrings.CozyBarClass)
   const sidebar = document.getElementById(DOMStrings.SidebarID)
   const rootModal = getRootModal()
   const immersive = Boolean(getFlagshipMetadata().immersive)
+  const openProp = open || false
 
-  useSetFlagshipUI(
+  useDialogSetFlagshipUI(
+    /** As per MUI spec, open is mandatory and defaults to false, adding it here in case of dev error somewhere */
+    openProp,
     makeOnMount({ fullscreen, theme, sidebar, rootModal, cozybar }),
     makeOnUnmount({ rootModal, theme, immersive, sidebar, cozybar }),
     makeCaller(!!fullscreen, !!rootModal, immersive)
   )
 }
 
+/**
+ * Custom version of useSetFlagshipUi() that is aware of the Dialog component.
+ *
+ * The difference here is that we send messages to the Native app when a props change.
+ * In the original version, we send the mount message as soon as the component is rendered.
+ *
+ * Dialog can be rendered but hidden, so we need to wait for the open prop to be true
+ */
+export const useDialogSetFlagshipUI = (
+  open: boolean,
+  onMount: FlagshipUI,
+  onUnmount?: FlagshipUI,
+  caller?: string
+): void => {
+  const webviewIntent = useWebviewIntent()
+
+  useEffect(() => {
+    if (open)
+      parseArg(webviewIntent, onMount, `${caller || 'unknown'} (onOpenMount)`)
+
+    return () => {
+      /**
+       * As we are listening to the open prop, we still want to send an unmount message when the prop changes to false.
+       * To avoid false positives, we need to ensure the component is currently showing.
+       * We do that by checking if value of open during this cleanup cycle is false,
+       * if it is, that means the component is currently showing and is in the process of hiding.
+       *
+       * Note that this will also handle abrupt unmounting, as in hiding the dialog without using the open prop.
```

Comment:
> Note that this will also handle abrupt unmounting, as in hiding the dialog without using the open prop

je crois qu'on ne peut pas utiliser de dialog sans prop `open`, si ? 🤔 Dans les app soit on force la prop open à true dans le comp qui a la dialog et on conditionne le montage du composant, soit (et ce cas pose plus de souci qu'autre chose donc on le fait rarement voire jamais) on monte toujours la dialog, et on change la prop `open` selon le besoin.

Reply:
J'ai fait en sorte d'appliquer le même fonctionnement à open === undefined et open === false du coup ✅